tarofreighter/chain_porter.go

```diff
@@ -650,27 +652,47 @@ func (p *ChainPorter) stateStep(currentPkg sendPackage) (*sendPackage, error) {
 		if err != nil {
 			return nil, err
 		}
-		senderScriptKey, err := p.cfg.KeyRing.DeriveNextKey(
-			ctx, tarogarden.TaroKeyFamily,
-		)
-		if err != nil {
-			return nil, err
-		}
 
-		// We'll assume BIP 86 everywhere, and use the tweaked key from
-		// here on out.
-		currentPkg.SenderScriptKey = asset.NewScriptKeyBIP0086(
-			senderScriptKey,
-		)
+		// If we are sending the full value of the input asset, we will
+		// need to create a split with unspendable change.
+		if inputAsset.Type == asset.Normal && !needsSplit {
+			emptyRootKey, err := btcec.ParsePubKey(
+				asset.NScriptKey[:],
+			)
+			if err != nil {
+				return nil, err
+			}
+
+			currentPkg.SenderScriptKey = asset.ScriptKey{
+				PubKey: emptyRootKey,
+				TweakedScriptKey: &asset.TweakedScriptKey{
+					RawKey: keychain.KeyDescriptor{
+						PubKey: emptyRootKey,
+					},
+				},
+			}
+		} else {
+			senderScriptKey, err := p.cfg.KeyRing.DeriveNextKey(
+				ctx, tarogarden.TaroKeyFamily,
+			)
+			if err != nil {
+				return nil, err
+			}
+
+			// We'll assume BIP 86 everywhere, and use the tweaked key from
+			// here on out.
+			currentPkg.SenderScriptKey = asset.NewScriptKeyBIP0086(
+				senderScriptKey,
+			)
+		}
 
 		// If we need to split (addr amount < input amount), then we'll
 		// transition to prepare the set of splits. If not,then we can
 		// assume the splits are unnecessary.
 		//
 		// TODO(roasbeef): always need to split anyway see:
 		// https://github.com/lightninglabs/taro/issues/121
-		currentPkg.NeedsSplit = needsSplit
-		if needsSplit {
+		if inputAsset.Type == asset.Normal {
```

Comment:
👍 

Once we add collectibles, we can circle back and refactor this a bit (can remove that extra state for now, since we're handling mainly t he async send process rn).